### ISSUE TITLE
Style link and asset order is not fixed

### ISSUE THREAD

**Issue by ImrePyhvel** [T4MVC/T4MVC#111]
When multiple developers are working on project and they have different cultures with different alphabet order then generated file keeps constantly reordering generated lines. This is annoying and causes junk-changes in VCS.

For example, the english order is `..rstuvwxyZ`, while estonian has `..rsšZžtuvw..`. Obviously this occurs for other chars as well and for many other culture pairs.

Generated items should be ordered by invariant culture everywhere.

**Comment by davidebbo** [comment]
Makes sense. Want to take a crack at a PR?

**Comment by ImrePyhvel** [comment]
Poked around a bit to see what's involved.. 

**Scope change?**
Currently there does not seem to be any explicit order anywhere and it's relying on the order DTE is returning its child ProjectItems. So the ordering culture can't be "changed", it has to be "added" ;)
Probably the same ordering logic should be applied to not just styles but to everything there is - areas, controllers, views, links, assets, .. 

**Desired order?**
It is not obvious what is the best fixed order. Should it be InvariantCulture? InvariantCultureIgnoreCase? Ordinal? OrdinalIgnoreCase?  Also there's the choice of whether to order names before or after sanitation. 

The current effective order (inherited from DTE) seems to be case-insenstive sort based on pre-sanitized names. So the generated C# names do not end up alphabetically ordered (not that it's a requirement or a problem for practical purposes, except for occasional users with OCD). I'd like to make it ordered but it would cause a one-time partial shuffle of rows in the generated file for almost all users.

Thoughts?

**Comment by davidebbo** [comment]
Yeah, I had quickly scanned through the code, hoping to find an OrderBy, and indeed there isn't one. :(

You ask some good questions about the desired order. OrdinalIgnoreCase seems like best choice. I would say that if at all possible, minimizing the churn on existing generated files would be a good goal. But if that proves too quirky and what makes the most sense causes a bunch of things to move around (just one time), then maybe it's not that terrible.
